ppb/systems/__init__.py

```diff
@@ -78,14 +98,14 @@ def render_background(self, scene):
         self.window.fill(scene.background_color)
 
     def prepare_resource(self, game_object):
-        image_name = game_object.__image__()
-        if image_name is flags.DoNotRender:
+        image = game_object.__image__()
+        if image is flags.DoNotRender:
             return None
-        image_name = str(image_name)
-        if image_name not in self.resources:
-            self.register_renderable(game_object)
+        if isinstance(image, str):
+            logger.warn(f"Using string resources is deprecated, use ppb.Image instead. Got {image!r}")
```

Comment:
👍

Reply:
Only thing about this is that it's extremely verbose (every frame).

Reply:
Could throw in a tracking variable and only throw it once? (Or once per object?) Would `lru_cache` wrapped around a print statement make it run once per object passed?

Reply:
Yeah, all of those things. It's just annoying, though?